app/src/main/java/com/alexchan/wallpaper/MainActivity.kt

```diff
@@ -1,12 +1,124 @@
 package com.alexchan.wallpaper
 
-import androidx.appcompat.app.AppCompatActivity
 import android.os.Bundle
+import android.util.Log
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.view.GravityCompat
+import androidx.navigation.fragment.NavHostFragment
+import com.google.android.material.bottomnavigation.BottomNavigationView
+import com.google.android.material.navigation.NavigationView
+import kotlinx.android.synthetic.main.activity_main.*
+
+class MainActivity : AppCompatActivity(), View.OnClickListener {
 
-class MainActivity : AppCompatActivity() {
+//    private val REQUEST_CODE = 100
+
+    // For Log statement
+    companion object {
+        private val TAG: String?= MainActivity::class.simpleName
+    }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
+
+        // Set TopToolbar OnClickListener
+        topToolbar.setOnClickListener{openNavDrawer()}
+
+        // Set NavigationView Drawer OnItemCLickListener
+        navigation_view.setNavigationItemSelectedListener(drawerNavListener())
+        // Set Checked Item Wallpaper by default
+        navigation_view.setCheckedItem(R.id.drawer_Item_Wallpaper)
+
+        // Set BottomNavigationView OnItemClickListener
+        bottom_navigation.setOnNavigationItemSelectedListener(bottomNavListener())
+
+        // Default Wallpaper Fragment
+        Log.d(TAG, "Wallpaper fragment host created by default")
+        createNavigationHost(R.navigation.navigation_host_wallpaper)
+    }
+
+    // Open Navigation Drawer
+    private fun openNavDrawer() {
+        Log.d(TAG, "Drawer is Open")
+        drawer_layout.openDrawer(GravityCompat.START)
+    }
+
+    // Create Navigation Host
+    private fun createNavigationHost(navigationHost: Int) {
+        val host = NavHostFragment.create(navigationHost)
+        supportFragmentManager.beginTransaction().replace(R.id.mainNavHostFragment, host)
+            .setPrimaryNavigationFragment(host).commit()
+    }
+
+    // Bottom Navigation Listener
+    private fun bottomNavListener() : BottomNavigationView.OnNavigationItemSelectedListener =
+        BottomNavigationView.OnNavigationItemSelectedListener {
+
+            when(it.itemId) {
+                R.id.item_Wallpaper -> {
```

Comment:
Naming is wrong, use either camel case or underscore with all small capital letters